src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -1,87 +1,24 @@
-import React, { useState, useMemo, useRef, useCallback } from "react";
-import { useHistory } from "react-router-dom";
-import { OverlayTrigger, Popover } from "react-bootstrap";
+import React, { ReactNode } from "react";
 
-import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { faTicketAlt } from "@fortawesome/free-solid-svg-icons";
+import { currentVenueSelectorData, parentVenueSelector } from "utils/selectors";
 
-import firebase from "firebase/app";
-
-import { DEFAULT_PROFILE_IMAGE, PLAYA_VENUE_ID } from "settings";
-import { IS_BURN } from "secrets";
-
-import { UpcomingEvent } from "types/UpcomingEvent";
-
-import {
-  currentVenueSelectorData,
-  parentVenueSelector,
-  radioStationsSelector,
-} from "utils/selectors";
-import { hasElements } from "utils/types";
-import { venueInsideUrl } from "utils/url";
-
-import { useRadio } from "hooks/useRadio";
 import { useSelector } from "hooks/useSelector";
-import { useUser } from "hooks/useUser";
-import { useVenueId } from "hooks/useVenueId";
 import { useFirestoreConnect } from "hooks/useFirestoreConnect";
 
-import { GiftTicketModal } from "components/organisms/GiftTicketModal/GiftTicketModal";
-import { ProfilePopoverContent } from "components/organisms/ProfileModal";
-import { RadioModal } from "components/organisms/RadioModal/RadioModal";
-import { SchedulePageModal } from "components/organisms/SchedulePageModal/SchedulePageModal";
-
-import NavSearchBar from "components/molecules/NavSearchBar";
-import UpcomingTickets from "components/molecules/UpcomingTickets";
-import { VenuePartygoers } from "components/molecules/VenuePartygoers";
-
-import { NavBarLogin } from "./NavBarLogin";
-
 import "./NavBar.scss";
-import * as S from "./Navbar.styles";
 import "./playa.scss";
 
-const TicketsPopover: React.FC<{ futureUpcoming: UpcomingEvent[] }> = (
-  props: unknown,
-  { futureUpcoming }
-) => (
-  <Popover id="popover-basic" {...props}>
-    <Popover.Content>
-      <UpcomingTickets events={futureUpcoming} />
-    </Popover.Content>
-  </Popover>
-);
-
-const ProfilePopover = (
-  <Popover id="profile-popover">
-    <Popover.Content>
-      <ProfilePopoverContent />
-    </Popover.Content>
-  </Popover>
-);
-
-const GiftPopover = (
-  <Popover id="gift-popover">
-    <Popover.Content>
-      <GiftTicketModal />
-    </Popover.Content>
-  </Popover>
-);
-
 interface NavBarPropsType {
-  redirectionUrl?: string;
-  hasBackButton?: boolean;
+  leftComponent?: ReactNode;
+  rightComponent?: ReactNode;
 }
 
 const NavBar: React.FC<NavBarPropsType> = ({
-  redirectionUrl,
-  hasBackButton = true,
+  leftComponent,
+  rightComponent,
 }) => {
```

Comment:
We should name these props better.

Reply:
The internal styles of the component split these props in to left and right corners of the screen. I don't really have any other ideas for better generic names. Feel free to suggest something.

Reply:
In one of my other comments I talked about keeping the navbar-links container within the navbar component, as it defines its ‘slot’, so that would probably be named similarly.

I would have to look closer at the code/usage of these to pick better names for these.

Reply:
I'm not okey with putting name specific class names in a component that's accepting props which can be anything. If someone wants to pass something else tomorrow there is a high chance that he'd have to change the navbar-links container too.

Reply:
Which is why I said to name the slot the same as the container it’s going into..?

@mike-lvov do you understand what I’m saying here? Maybe you’ll be able to explain it better than I seem to be able to..

Reply:
I guess there is a small misunderstanding here.

> I'm not okey with putting name specific class names in a component that's accepting props which can be anything. If someone wants to pass something else tomorrow there is a high chance that he'd have to change the navbar-links container too.

Totally agree with you. We want the navbar to be as versatile as it's possible.

I guess what was meant is that `leftSlot, rightSlot`(which don't have any content connotation. They can contain anything) are a bit more diverse/logical names than `leftComponent, rightComponent`.

Personally, I don't have a very strong opinion on this. Both work for me

Reply:
Nah, @mike-lvov is missing my point here also.

In this refactor the code now looks essentially this:

`src/components/molecules/NavBar/NavBar.tsx`
```typescript
export interface NavBarPropsType {
  leftComponent?: ReactNode;
  rightComponent?: ReactNode;
}

// ..snip..

export const NavBar: React.FC<NavBarPropsType> = ({
  leftComponent,
  rightComponent,
}) => {

  // ..snip..

  return (
    // ..snip..
    <div>
      {leftComponent}
      {rightComponent}
    </div>
    // ..snip..
```

Which ends up shunting everything else to the calling components in an unstructured way:

`src/components/organisms/WithNavigationBar/WithNavigationBar.tsx`
```typescript
  const leftComponent = (
    <div className="nav-logos">
      <div className="nav-sparkle-logo" onClick={navigateToHomepage}>
        <div />
      </div>
      <div className="nav-sparkle-logo_small">
        <div />
      </div>
      {venue && (
        <>
          <NavbarSchedule />
          <VenuePartygoers />
        </>
      )}
    </div>
  );

  const rightComponent = user ? (
    <div className="navbar-links">
This conversation was marked as resolved by denisdimitrov
      {venue && <NavSearchBar />}

      {venue?.showGiftATicket && (
        <OverlayTrigger
          trigger="click"
          placement="bottom-end"
          rootClose={true}
          overlay={
            <Popover id="gift-popover">
              <Popover.Content>
                <GiftTicketModal />
              </Popover.Content>
            </Popover>
          }
        >
          <span className="private-chat-icon">
            <div className="navbar-link-gift" />
          </span>
        </OverlayTrigger>
      )}

      <NavbarRadio />
      <NavbarProfile />
    </div>
    <Footer />
  </>
);
  ) : (
    <NavBarLogin />
  );
```

This unnecessarily moves the 'layout concerns' outside of the NavBar. Things that consume the `NavBar` shouldn't need to think about it's internal classes/structure/etc. They should just consume a simple 'public interface' that is exposed through the props. That's what I mean when I talk about 'slots'.

As I believe I have explained deeper in previous review comments, the contains sections that are 'always there' (and thus belong in the `NavBar` itself, and 'only sometimes wanted', in which case they should be injected through these 'slots'.

So looking at the above example, the `.navbar-links` `div` is a part of the `NavBar`, and it should exist within the `NavBar`; it is a 'slot' to inject content into. Static content that should always appear there (eg. `NavbarProfile`) should also just be added directly to the `NavBar` component. Things that are only sometimes needed (eg. `NabarRadio`, etc) would then be 'injected' into this slot from the calling location that needs them.

Similarly, `.nav-logos`/`.nav-sparkle-logo_small`/etc all belongs within the `NavBar` component itself, as it is 'static content' that 'belongs' to the `NavBar`. Whereas the `NavbarSchedule`/etc is stuff that only sometimes needs to be there, so it should be injected via the slot.

The changes i'm asking for here would take less time to implement than i've taken to explain them. They're not hard to do, but they are fundamentally a better and more correct solution than how it is currently implemented.

Reply:
> The changes i'm asking for here would take less time to implement than i've taken to explain them.

I'm not entirely sure why you asked for these changes in the first place. The whole slots concept seems weird in a header component, IMO all of it's children should be static components. renderProps are used when you need to pass dynamic components in different cases, we have a single header and it's rendered in a single place. This whole thing is unnecessary. Rendering the conditional components via slots, is an 'invisible' rule which makes no sense. I haven't seen this anywhere.

Don't get me wrong, I will still do it and apply the changes but this approach is confusing and weird.

Reply:
The header is rendere in 2 places. Admin and not-admin. And it should have different elements in it in those 2 locations; as I described above (eg. Radio, etc)

Reply:
Yes, the header is used in shown in 2 places but the component itself is rendered only in `WithNavigationBar`  which doesn't accept any props to change these "slots".